rotkehlchen/data_import/importers/rotki_events.py

```diff
@@ -0,0 +1,122 @@
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
+        events: List[HistoryBaseEntry] = []
+        identifier = csv_row['ID']
+        timestamp = TimestampMS(
+            deserialize_timestamp_from_date(
+                date=csv_row['Timestamp'],
+                formatstr=timestamp_format,
+                location='Rotki generic events importer',
+            ) * 1000,
+        )
+        fee = Fee(deserialize_asset_amount(csv_row['Fee'])) if csv_row['Fee'] else Fee(ZERO)  # noqa: E501
+        fee_currency = (
+            symbol_to_asset_or_token(csv_row['Fee Currency'])
+            if csv_row['Fee Currency'] and fee is not None else None
+        )
+        history_event = HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),
+            sequence_index=csv_row['sequence_index'],
+            timestamp=timestamp,
+            location=Location.EXTERNAL,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=symbol_to_asset_or_token(csv_row['Currency']),
+            balance=Balance(
+                amount=deserialize_asset_amount(csv_row['Amount']),
+                usd_value=ZERO,
+            ),
+            notes=csv_row['Description'],
+        )
+        events.append(history_event)
+        if fee != ZERO and fee_currency is not None:
+            fee_event = HistoryBaseEntry(
+                event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),  # noqa: E501
+                sequence_index=csv_row['sequence_index'] + 1,
+                timestamp=timestamp,
+                location=Location.EXTERNAL,
+                event_type=event_type,
+                event_subtype=HistoryEventSubType.FEE,
+                asset=fee_currency,
+                balance=Balance(
+                    amount=fee,
+                    usd_value=ZERO,
+                ),
+                notes=csv_row['Description'],
+            )
+            events.append(fee_event)
+        self.add_history_events(cursor, events)
+
+    def _import_csv(self, cursor: DBCursor, filepath: Path, **kwargs: Any) -> None:
+        """May raise:
+        - InputError if one of the rows is malformed
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for idx, row in enumerate(data):
+                try:
+                    # this is a requirement for history event entries
+                    row['sequence_index'] = idx
+                    self._consume_rotki_events(cursor, row, **kwargs)
```

Comment:
As for what I understood all the events are one line events. You could check that the identifier of the row wasn't seen before